static/app/components/discover/performanceCardTable.tsx

```diff
@@ -0,0 +1,667 @@
+import {getDuration} from 'app/utils/formatters';
+import {TableColumn} from 'app/views/eventsV2/table/types';
+import {GridCell, GridCellNumber} from 'app/views/performance/styles';
+import {TrendsDataEvents} from 'app/views/performance/trends/types';
+import {DiscoverField} from 'app/views/releases/detail/overview/index';
+import {performanceCardLabels} from 'app/views/releases/detail/utils';
+
+const FRONTEND_PLATFORMS: string[] = [...frontend, ...mobile];
+const BACKEND_PLATFORMS: string[] = [...backend, ...serverless];
+
+type ReleaseTableDataRow = {
+  [key: string]: React.ReactText;
+};
+
+export type ReleaseTableData = {
+  data?: Array<ReleaseTableDataRow>;
+  meta?: MetaType;
+};
+
+type Props = {
+  eventView: EventView;
+  releaseEventView: EventView;
+  organization: Organization;
+  project: ReleaseProject;
+  location: Location;
+  isLoading: boolean;
+  tableData: TableData | TrendsDataEvents | null;
+  releaseTableData: ReleaseTableData | null;
+  columnOrder: TableColumn<React.ReactText>[];
+  generateLink?: Record<
+    string,
+    (
+      organization: Organization,
+      tableRow: TableDataRow,
+      query: Query
+    ) => LocationDescriptor
+  >;
+};
+
+class PerformanceCardTable extends React.PureComponent<Props> {
+  renderRow(
+    row: TableDataRow | object,
+    rowIndex: number,
+    columnOrder: TableColumn<React.ReactText>[],
+    tableMeta: MetaType
+  ): React.ReactNode[] {
+    const {organization, location} = this.props;
+
+    const resultsRow = columnOrder.map((column, index) => {
+      const field = String(column.key);
+      // TODO add a better abstraction for this in fieldRenderers.
+      const fieldName = getAggregateAlias(field);
+      const fieldType = tableMeta[fieldName];
+
+      const fieldRenderer = getFieldRenderer(field, tableMeta);
+      let rendered = fieldRenderer(row, {organization, location});
+
+      const isNumeric = ['integer', 'number', 'duration'].includes(fieldType);
+      const key = `${rowIndex}:${column.key}:${index}`;
+      rendered = isNumeric ? (
+        <GridCellNumber>{rendered}</GridCellNumber>
+      ) : (
+        <GridCell id="border">{rendered}</GridCell>
+      );
+
+      return <BodyCellContainer key={key}>{rendered}</BodyCellContainer>;
+    });
+
+    return resultsRow;
+  }
+
+  renderChangeRow(row: object) {
+    const cells: any[] = [];
+
+    Object.values(row).forEach(r => cells.push(r));
+
+    const rendered = (
+      <BodyCellContainer id="changeBody">
+        {cells.map(cell => (
+          <GridCellNumber key={cell} id={!cell ? 'changeBorder' : 'changeNoBorder'}>
+            <StyledIconArrow
+              direction={cell > 0 ? 'up' : 'down'}
+              color={cell > 0 ? 'red300' : 'green300'}
+              size="xs"
+            />
+            {cell.includes('-')
+              ? cell.replace('-', '')
+              : cell.includes('NaN')
+              ? cell.replace('NaN', '0.00')
+              : cell}
```

Comment:
![image](https://user-images.githubusercontent.com/9060071/127315009-0b2cbc78-e7ac-4919-9f98-f91ccd23eb2b.png)

We should format this